File: packages/intune/ios/Plugin/Classes/Results/RemediateComplianceResult.swift
import Foundation
import Capacitor

@objc public class RemediateComplianceResult: NSObject, Result {
    let errorMessage: String?
    let errorTitle: String?
    let status: String

    init(errorMessage: String?, errorTitle: String?, status: String) {
        self.errorMessage = errorMessage
        self.errorTitle = errorTitle
        self.status = status
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["errorMessage"] = errorMessage == nil ? NSNull() : errorMessage
        result["errorTitle"] = errorTitle == nil ? NSNull() : errorTitle
        result["status"] = status
        return result as AnyObject
    }
}

File: packages/intune/android/src/main/java/io/capawesome/capacitorjs/plugins/intune/classes/results/RemediateComplianceResult.java
package io.capawesome.capacitorjs.plugins.intune.classes.results;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.intune.interfaces.Result;
import org.json.JSONObject;

public class RemediateComplianceResult implements Result {

    @Nullable
    private final String errorMessage;

    @Nullable
    private final String errorTitle;

    @NonNull
    private final String status;

    public RemediateComplianceResult(@Nullable String errorMessage, @Nullable String errorTitle, @NonNull String status) {
        this.errorMessage = errorMessage;
        this.errorTitle = errorTitle;
        this.status = status;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("errorMessage", errorMessage == null ? JSONObject.NULL : errorMessage);
        result.put("errorTitle", errorTitle == null ? JSONObject.NULL : errorTitle);
        result.put("status", status);
        return result;
    }
}
